### PR TITLE
fix pinning status for shuttle contents

### PR DESCRIPTION
**Pull request by en0ma**
Currently, when shuttles add content and fail to signal estuary about pin completion (maybe due to downtime), estuary will not check if this content is pinned. This leaves the content in a stale state on Estuary, but on the shuttle, it will be pinned and active.

This PR correctly sets the pin status on Estuary side when the content is created, and if any downtime occurs, when Estuary is back up, it will retry the pinning and be updated accordingly.

see here for more context https://filecoinproject.slack.com/archives/C033QS35MFA/p1659047543608269

### CODE REVIEW

File: cmd/estuary-shuttle/main.go [hunk 1]
```diff
@@ -1149,12 +1149,8 @@ func (s *Shuttle) handleLogLevel(c echo.Context) error {
 func (s *Shuttle) handleAdd(c echo.Context, u *User) error {
 	ctx := c.Request().Context()
 
-	if u.StorageDisabled || s.disableLocalAdding {
-		return &util.HttpError{
-			Code:    http.StatusBadRequest,
-			Reason:  util.ERR_CONTENT_ADDING_DISABLED,
-			Details: "uploading content to this node is not allowed at the moment",
-		}
+	if err := util.ErrorIfContentAddingDisabled(u.StorageDisabled || s.disableLocalAdding); err != nil {
```
Comment (gmelodie) [inline cmd/estuary-shuttle/main.go:1152]:
```suggestion
	if err := util.ErrorIfContentAddingDisabled(s.isContentAddingDisabled(u)); err != nil {
```

**Review by gmelodie — CHANGES_REQUESTED** [review]

File: cmd/estuary-shuttle/main.go (hunk 1 above)
Comment (en0ma) [reply cmd/estuary-shuttle/main.go:1152]: shuttle cant use `isContentAddingDisabled`

Comment (gmelodie) [reply cmd/estuary-shuttle/main.go:1152]: We could just add the function:

```
func (s *Shuttle) isContentAddingDisabled(u *User) bool {
	return s.disableLocalAdding || u.StorageDisabled
}
```

Comment (en0ma) [reply cmd/estuary-shuttle/main.go]: thank you

**Review by gmelodie — APPROVED** [review]

**Pull request merged**